### PR TITLE
CRM-17120 - Use CRM_Core_Pseudoconstant to lookup option labels

### PR DESCRIPTION
**Pull request by colemanw**
https://issues.civicrm.org/jira/browse/CRM-17120

### CODE REVIEW

**Comment by colemanw** [comment]
@eileenmcnaughton I haven't tested this out, can you?

File: CRM/Financial/Page/AJAX.php [hunk 1]
```diff
@@ -207,7 +207,7 @@ public static function assignRemove() {
           case 'remove':
             $recordPID = CRM_Core_DAO::getFieldValue('CRM_Financial_DAO_FinancialTrxn', $recordID, 'payment_instrument_id');
             $batchPID = CRM_Core_DAO::getFieldValue('CRM_Batch_DAO_Batch', $entityID, 'payment_instrument_id');
-            $paymentInstrument = CRM_Core_OptionGroup::getLabel('payment_instrument', $batchPID);
+            $paymentInstrument = CRM_Core_PseudoConstant::getLabel('CRM_Batch_DAO_Batch', 'payment_instrument', $batchPID);
```
Comment (eileenmcnaughton) [inline CRM/Financial/Page/AJAX.php]: Is it better practice to pass in the BAO?

Comment (colemanw) [inline CRM/Financial/Page/AJAX.php]: yes, good point

**Comment by colemanw** [comment]
PR ammended

**Comment by eileenmcnaughton** [comment]
Actually I still think it's wrong - on the ones I've tacked it would need to be payment_instrument_id not payment instrument

https://github.com/civicrm/civicrm-core/pull/6658

(NB I added some unit tests on those ones - although I might not manage that for all 20 remaining

**Comment by colemanw** [comment]
Glad you checked! PR amended again.

**Comment by eileenmcnaughton** [comment]
That looks OK - I don't actually know how to trigger that line of code - but am happy to merge it on the basis of having visually reviewed it since it's going into master.

**Pull request merged**